halo/attest/keeper/keeper_test.go

```diff
@@ -0,0 +1,69 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/omni-network/omni/halo/attest/keeper"
+	"github.com/omni-network/omni/halo/attest/types"
+
+	k1 "github.com/cometbft/cometbft/crypto/secp256k1"
+	cmttypes "github.com/cometbft/cometbft/types"
+
+	ormv1alpha1 "cosmossdk.io/api/cosmos/orm/v1alpha1"
+	"cosmossdk.io/orm/model/ormdb"
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeeper_Add(t *testing.T) {
+	t.Parallel()
+
+	k, store, ctx := setupKeeper(t)
+
+	var votes []*types.AggVote
+	fuzz.New().NilChance(0).NumElements(1, 3).Fuzz(&votes)
+	msg := &types.MsgAddVotes{
+		Authority: "test-authority",
+		Votes:     votes,
+	}
+
+	err := k.Add(ctx, msg)
+	require.NoError(t, err)
+
+	schema := &ormv1alpha1.ModuleSchemaDescriptor{SchemaFile: []*ormv1alpha1.ModuleSchemaDescriptor_FileEntry{
+		{Id: 1, ProtoFileName: keeper.File_halo_attest_keeper_attestation_proto.Path()},
+	}}
+
+	modDB, err := ormdb.NewModuleDB(schema, ormdb.ModuleDBOptions{KVStoreService: store})
+	require.NoError(t, err)
+
+	attstore, err := keeper.NewAttestationStore(modDB)
+	require.NoError(t, err)
+
+	a, err := attstore.AttestationTable().Get(ctx, 1)
+	require.NoError(t, err)
+	require.NotNil(t, a)
```

Comment:
you're going to have to do a lot more validation here. We need to ensure the data is 100% correct. You also need to verify signatures table.